### PR TITLE
Add handlers option to downloadTool

### PR DESCRIPTION
**Pull request by RobiNino**
Adding an optional input of http client handlers to the downloadTool to allow an authenticated download.

### CODE REVIEW

File: tool.ts [hunk 1]
```diff
@@ -194,10 +193,13 @@ export function findLocalToolVersions(toolName: string, arch?: string) {
  *
  * @param url       url of tool to download
  * @param fileName  optional fileName.  Should typically not use (will be a guid for reliability). Can pass fileName with an absolute path.
+ * @param handlers  optional handlers array.  Handlers to pass to the HttpClient for the tool download.
```
Comment (damccorm) [inline tool.ts]: nit: Make it clear handlers are for auth, e.g.

`Auth handlers to pass to the HttpClient for the tool download`

File: tool.ts [hunk 2]
```diff
@@ -521,6 +523,7 @@ function _createExtractFolder(dest?: string): string {
  * @param regex     regex to use for version matches
  */
 export async function scrape(url: string, regex: RegExp): Promise<string[]> {
+    let http: httpm.HttpClient = new httpm.HttpClient(userAgent, null, requestOptions);
```
Comment (damccorm) [inline tool.ts]: Any reason not to allow auth handlers here as well?

File: test/tests/toolTests.ts [hunk 3]
```diff
@@ -171,4 +172,38 @@ describe('Tool Tests', function () {
             }
         });
     });
+
+    it('download with basic authentication', function () {
+        this.timeout(2000);
+
+        return new Promise<void>(async (resolve, reject) => {
+            // General parameters:
+            let username = "usr";
```
Comment (damccorm) [inline test/tests/toolTests.ts]: Nit: add types to variables here

File: tool.ts (hunk 2 above)
Comment (RobiNino) [reply tool.ts]: Not quite, wasn't sure if it is a desired functionality. Added the option now

**Comment by damccorm** [comment]
Cool, this looks great. I'll merge once the cla is signed!

**Comment by RobiNino** [comment]
Done, Thanks @damccorm !

**Pull request merged**